subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/moduleconverter/DefaultRootComponentMetadataBuilderTest.groovy

```diff
@@ -137,6 +137,28 @@ class DefaultRootComponentMetadataBuilderTest extends Specification {
         ]
     }
 
+    def "discards component metadata when hierarchy changes"() {
+        componentIdentifierFactory.createComponentIdentifier(_) >> {
+            new DefaultModuleComponentIdentifier(mid, '1.0')
+        }
+        def root = builder.toRootComponentMetaData()
+
+        def conf = root.getConfiguration("conf")
+        assert conf.needsReevaluate()
+        conf.realizeDependencies()
+        assert !conf.needsReevaluate()
```

Comment:
🧐 Nitpick: This doesn't really follow the given-when-then pattern with assertions in the given part.

Reply:
The whole test is like this. I'm going to leave this as-is